numba/tests/doc_examples/test_interval_example.py

```diff
@@ -106,21 +106,69 @@ def impl_interval(context, builder, sig, args):
 
         # magictoken.interval_unbox.begin
         from numba.extending import unbox, NativeValue
+        from contextlib import contextmanager, ExitStack
+
+        @contextmanager
+        def early_exit_if(builder, stack: ExitStack, cond):
+            """
+            Emit code similar to::
+
+                if (cond) {
+                    <body>
+                    return;
+                }
+                <everything after this call>
+
+            However, this "return" will just break out of the current `ExitStack`,
+            rather than out of the whole function
+            """
+            then, otherwise = stack.enter_context(builder.if_else(cond, likely=False))
+            with then:
+                yield
+            stack.enter_context(otherwise)
+
+
+        def early_exit_if_null(builder, stack, obj):
+            return early_exit_if(builder, stack, cgutils.is_null(builder, obj))
+
 
         @unbox(IntervalType)
         def unbox_interval(typ, obj, c):
             """
             Convert a Interval object to a native interval structure.
             """
-            lo_obj = c.pyapi.object_getattr_string(obj, "lo")
-            hi_obj = c.pyapi.object_getattr_string(obj, "hi")
-            interval = cgutils.create_struct_proxy(typ)(c.context, c.builder)
-            interval.lo = c.pyapi.float_as_double(lo_obj)
-            interval.hi = c.pyapi.float_as_double(hi_obj)
-            c.pyapi.decref(lo_obj)
-            c.pyapi.decref(hi_obj)
-            is_error = cgutils.is_not_null(c.builder, c.pyapi.err_occurred())
-            return NativeValue(interval._getvalue(), is_error=is_error)
+            ret_ptr = cgutils.alloca_once(c.builder, c.context.get_value_type(typ))
+            is_error_ptr = cgutils.alloca_once_value(c.builder, cgutils.false_bit)
+            fail_obj = c.context.get_constant_null(typ)
```

Comment:
In @kc611's PR, this `fail_obj` didn't exist. It's been a white since I wrote this, so I can't remember if this was required to avoid a bug, or if I was just being defensive.

In particular, I think the way this is written now means that `NativeValue.value` contains an expression that dereferences a null pointer if there is an error; but this is arguably a good thing, because nothing should be using `NativeValue.value` in the first place if `is_error` is set.

Reply:
> In particular, I think the way this is written now means that `NativeValue.value` contains an expression that dereferences a null pointer if there is an error; but this is arguably a good thing, because nothing should be using `NativeValue.value` in the first place if `is_error` is set.

I am not sure what you mean here, can you elaborate?

Reply:
I've just reworked this code so that the comment is no longer relevant.